packages/data-context/src/codegen/spec-options.ts

```diff
@@ -6,10 +6,18 @@ import type { CodeGenType } from '@packages/graphql/src/gen/nxs.gen'
 interface CodeGenOptions {
   codeGenPath: string
   codeGenType: CodeGenType
-  specFileExtension: string
   erroredCodegenCandidate?: string | null
 }
 
+// Spec file extensions that we will preserve when updating the file name
+// due the existence of duplicate files.
+//
+// Example:
+//   Button.cy.ts   -> Button-copy-1.cy.ts
+//   Button_spec.js -> Button-copy-1_spec.js
+//   Button.foo.js  -> Button.foo-copy-1.js
+export const expectedSpecExtensions = ['.cy', '.spec', '.test', '-spec', '-test', '_spec']
```

Comment:
While the immediate problem is fixed, the existence of this set is problematic. It's still possible for users with a custom spec pattern to create a new file and have it not match their spec pattern after our '-copy-x' is injected. We're getting into edge-case, explicitly recommended against patterns here, but ideally it'd still work.

If we want this to be fool-proof, we'll need to detect duplicated filenames prior to submission so that the user can intelligently navigate this. Once we reach this point in the lifecycle, we're shooting in the dark.

Reply:
Do you want to open a task to track this improvement? I feel like we'll see a few reports about this...

Reply:
Logged https://cypress-io.atlassian.net/browse/UNIFY-1810